SCClassLibrary/Common/Core/Function.sc

```diff
@@ -72,12 +72,79 @@ Function : AbstractFunction {
 		// evaluate a function, using arguments from the supplied environment
 		// slightly faster than valueEnvir and does not replace the currentEnvironment
 		^this.valueArray(prototypeFrame)
+
+	valueWith {|argumentsArray=([]), keywordArgumentEnvir=(()), variableArgumentsArray=([])|
```

Comment:
> I figure that doesNotUnderstandWithKeys may just produce an array of (un-keyworded) argument values that is longer than the number of argument names. In this case, the keywordArgumentEnvir will be empty anyhow.

It isn't possible to have doesNotUnderstandWithKeys be called without any keyword args... unless the user does so themselves....

I will try and remove it and see what happens.

Reply:
I suppose it may happen that someone calls a message with keyword arguments but also supplies more positional arguments than argument names exist. Then, still `doesNotUnderstandWithKeys` would be called?

Reply:
I've removed it and it all seems to work.
Yes it would be called then, but so long as performWith or makePerformableArray is called, it will do the right thing.